core/src/main/java/com/datastax/oss/driver/internal/core/data/DefaultTupleValue.java

```diff
@@ -93,6 +95,44 @@ private void readObject(ObjectInputStream stream) throws InvalidObjectException
     throw new InvalidObjectException("Proxy required");
   }
 
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+
+    if (o == null || !(o instanceof TupleValue)) return false;
+    DefaultTupleValue that = (DefaultTupleValue) o;
+
+    if (this.protocolVersion() != that.protocolVersion()) return false;
+
+    for (int i = 0; i < values.length; i++) {
+      DataType innerThisType = type.getComponentTypes().get(i);
+      DataType innerThatType = type.getComponentTypes().get(i);
+      if (!innerThisType.equals(innerThatType)) {
+        return false;
+      }
+      Object thisValue =
+          this.codecRegistry()
+              .codecFor(innerThatType)
+              .decode(this.values[i], this.protocolVersion());
```

Comment:
I wonder if we should handle potential errors here: this can happen if the tuple value contains custom types but became detached from the session, so the codec lookup will fail. Maybe we should log a warning and fall back to byte array comparisons in that specific case. Or is it better to simply let the method throw?